### PR TITLE
Initial

### PR DESCRIPTION
**Pull request by michz**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,88 @@
...
+Role Variables
+--------------
+
+    # Install prerequisites via apt (disable and do it manually to use another package manager)
+    zammad_install_prerequisites: true
+
+    # If set to true, the setup process is triggered.
+    zammad_install: true
+
+    # If set to true, the configuration process is done.
```
Comment (sd-lueckel) [inline README.md]: Is this sentence correct? When i read it, it sounds like a description for an output => 'i have configured everything, now i am done!'

File: meta/main.yml [hunk 2]
```diff
@@ -0,0 +1,264 @@
+galaxy_info:
+  author: your name
```
Comment (sd-lueckel) [inline meta/main.yml]: Some information about solutionDrive would be nice

File: meta/main.yml [hunk 3]
```diff
@@ -0,0 +1,264 @@
...
+  # branch will be accepted. Otherwise, in all cases, the repo's default branch
+  # (usually master) will be used.
+  #github_branch:
+  
+  #
+  # Below are all platforms currently available. Just uncomment
+  # the ones that apply to your role. If you don't see your
+  # platform on this list, let us know and we'll get it added!
+  #
+  #platforms:
```
Comment (sd-lueckel) [inline meta/main.yml]: Configured platforms would be nice, too!

File: meta/main.yml [hunk 4]
```diff
@@ -0,0 +1,264 @@
...
+  #  - ascii
+  #  - beowulf
+  #  - ceres
+  #  - jessie
+  #- name: eos
+  #  versions:
+  #  - all
+  #  - Any
+  
+  galaxy_tags: []
```
Comment (sd-lueckel) [inline meta/main.yml]: Tags would be nice also

File: tasks/nginx_setup.yml [hunk 5]
```diff
@@ -0,0 +1,5 @@
+---
+- name: Install system prerequisites
+  apt:
```
Comment (sd-lueckel) [inline tasks/nginx_setup.yml]: In my opinion we should use 'package' instead of 'apt'

File: tasks/prerequisites.yml [hunk 6]
```diff
@@ -0,0 +1,34 @@
+---
+- name: Install system prerequisites
+  apt:
```
Comment (sd-lueckel) [inline tasks/prerequisites.yml]: In my opinion we should use 'package' instead of 'apt'

**Review by sd-lueckel — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (mattagohni) [reply README.md]: maybe like with the variable before `triggered` instead of `done` would do the trick...

File: tasks/zammad_setup.yml [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+  with_items:
+    - bundler
+    - rake
+    - rails
+
+- name: Install application dependencies with mysql database support
+  shell: bundle install --without test development postgres
+  args:
+    chdir: /opt/zammad
+  when: zammad_db_mysql
```
Comment (mattagohni) [inline tasks/zammad_setup.yml]: I think here `zammad_db_postgresql` should be tested instead of `zammad_db_mysql`

File: tasks/zammad_setup.yml [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+- name: Install base dependencies: bundler, rake and rails
+  gem:
+    name: "{{ item }}"
+    state: present
+  with_items:
+    - bundler
+    - rake
+    - rails
+
+- name: Install application dependencies with mysql database support
```
Comment (mattagohni) [inline tasks/zammad_setup.yml]: Also the name of the task doesn't fit waht it does (mysql vs. postgresql)

File: tasks/zammad_setup.yml [hunk 9]
```diff
@@ -0,0 +1,45 @@
...
+  shell: bundle install --without test development postgres
+  args:
+    chdir: /opt/zammad
+  when: zammad_db_mysql
+
+- name: Install dependencies with postgres database support
```
Comment (mattagohni) [inline tasks/zammad_setup.yml]: and here it is the other way around.. :)

File: tests/inventory [hunk 10]
```diff
@@ -0,0 +1 @@
+localhost
```
Comment (mattagohni) [inline tests/inventory]: blank line?

File: tests/test.yml [hunk 11]
```diff
@@ -0,0 +1,5 @@
+---
+- hosts: localhost
+  remote_user: root
+  roles:
+    - .
```
Comment (mattagohni) [inline tests/test.yml]: blank line?

**Review by mattagohni — CHANGES_REQUESTED** [review]
besides the minor issues, the role looks great. Would be awesome if we could get the mixed-up mysql/postgresql stuff sorted out.

File: tasks/nginx_setup.yml (hunk 5 above)
Comment (michz) [reply tasks/nginx_setup.yml]: Are you sure that the same package names work?

File: tasks/prerequisites.yml (hunk 6 above)
Comment (michz) [reply tasks/prerequisites.yml]: Are you sure that the same package names work?

File: tasks/zammad_setup.yml (hunk 7 above)
Comment (michz) [reply tasks/zammad_setup.yml]: No. See: "without"-Syntax!
https://docs.zammad.org/en/latest/install-source.html#for-postgresql-note-the-option-says-without-mysql

File: tasks/zammad_setup.yml (hunk 8 above)
Comment (michz) [reply tasks/zammad_setup.yml]: No. See: "without"-Syntax!
https://docs.zammad.org/en/latest/install-source.html#for-postgresql-note-the-option-says-without-mysql

File: tasks/zammad_setup.yml (hunk 9 above)
Comment (michz) [reply tasks/zammad_setup.yml]: No. See: "without"-Syntax!
https://docs.zammad.org/en/latest/install-source.html#for-postgresql-note-the-option-says-without-mysql

File: tasks/nginx_setup.yml (hunk 5 above)
Comment (sd-lueckel) [reply tasks/nginx_setup.yml]: Yes, because 'package' is only a wrapper around. For example: On ubuntu ansible will use apt in the background

Comment (sd-lueckel) [reply tasks/nginx_setup.yml]: If the platform under 'meta/main.yml' is ubuntu only it is ok for me to leave it as 'apt' for the moment

File: tasks/zammad_setup.yml (hunk 8 above)
Comment (mattagohni) [reply tasks/zammad_setup.yml]: oh I see... :-) but that's weird... :-D

File: tasks/zammad_setup.yml [hunk 12]
```diff
@@ -0,0 +1,45 @@
...
+  shell: bundle install --without test development mysql
+  args:
+    chdir: /opt/zammad/
+  when: zammad_db_mysql
```
Comment (mattagohni) [inline tasks/zammad_setup.yml]: but here is it the wrong variable, isn't it? :) So the tested variable should be `zammad_db_postgresql` if I'm not missing something again.

File: tasks/zammad_setup.yml (hunk 7 above)
Comment (mattagohni) [reply tasks/zammad_setup.yml]: makes sense then

File: tasks/zammad_setup.yml (hunk 8 above)
Comment (michz) [reply tasks/zammad_setup.yml]: I think this is one of the reasons the word "Brainfuck" was invented. I agree, the inverted logic is not good ...

File: tasks/prerequisites.yml [hunk 13]
```diff
@@ -0,0 +1,34 @@
+---
+- name: Install system prerequisites
+  packages:
```
Comment (mattagohni) [inline tasks/prerequisites.yml]: the module is `package` instead of `packages`.

File: tests/inventory (hunk 10 above)
Comment (michz) [reply tests/inventory]: Done.

**Review by mattagohni — DISMISSED** [review]

**Review by sd-lueckel — DISMISSED** [review]

**Comment by michz** [comment]
No more Work In Progress, ready to re-review!
Role works for setup and configuration.

**Review by sd-lueckel — APPROVED** [review]

**Comment by sd-lueckel** [comment]
Nice work!

**Review by twojtylak — APPROVED** [review]

**Pull request merged**